[Bug] [sink-Milvus] Cannot invoke "org.apache.seatunnel.api.table.factory.TableSinkFactoryContext.getOptions()" because "context" is null

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

在v2.3.13中，通过seatunnel 往Milvus中导入数据一直报错。

数据embedding的执行结果是符合预期的，但是从debug日志中看，获取不到 context 无法初始化sink 进而初始化一个错误的milvus 表

```
2026-05-11 18:03:56,970 DEBUG [o.a.s.a.t.f.FactoryUtil       ] [main] - java.lang.NullPointerException: Cannot invoke "org.apache.seatunnel.api.table.factory.TableSinkFactoryContext.getOptions()" because "context" is null
        at org.apache.seatunnel.connectors.seatunnel.milvus.sink.MilvusSinkFactory.createSink(MilvusSinkFactory.java:56)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.lambda$createAndPrepareSink$1(FactoryUtil.java:213)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.isFallback(FactoryUtil.java:458)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSink(FactoryUtil.java:209)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.createSinkAction(MultipleTableJobConfigParser.java:690)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSink(MultipleTableJobConfigParser.java:613)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:260)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:122)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:190)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:204)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:40)

2026-05-11 18:03:56,977 INFO  [o.a.s.a.t.f.FactoryUtil       ] [main] - Create sink 'Milvus' with upstream input catalog-table[database: null, schema: null, table: raw_data]
2026-05-11 18:03:56,979 INFO  [p.MultipleTableJobConfigParser] [main] - Unsupported multi table sink api, rollback to sink template
```
执行中的上下文也发生错乱
`2026-05-11 18:03:57,526 INFO  [a.s.a.s.SaveModeExecuteWrapper] [seatunnel-coordinator-service-2] - Executing save mode for table: data_goverment_test.raw_data, with SchemaSaveMode: ERROR_WHEN_SCHEMA_NOT_EXIST, DataSaveMode: APPEND_DATA using Catalog: S3File
2026-05-11 18:03:57,530 DEBUG [g.n.s.i.g.n.NettyClientHandler] [grpc-nio-worker-ELG-1-4] - [id: 0xe12e4b58, L:/172.20.3.24:51994 - R:/10.0.10.49:19530] OUTBOUND HEADERS: streamId=5 headers=GrpcHttp2OutboundHeaders[:authority: 10.0.10.49:19530, :path: /milvus.proto.milvus.MilvusService/HasCollection, :method: POST, :scheme: http, content-type: application/grpc, te: trailers, user-agent: grpc-java-netty/1.59.1, authorization: cm9vdDpNaWx2dXM=, dbname: default, grpc-accept-encoding: gzip] streamDependency=0 weight=16 exclusive=false padding=0 endStream=false`


进而报错
`The sink table not exist`

### SeaTunnel Version

2.3.13

### SeaTunnel Config

```conf
env {
  job.mode = "BATCH"
  parallelism = 1
  job.name = "complex_json_from_minio_to_es"
}

# 第一步：从MinIO读取整个JSON文件作为单行文本数据
source {
  # 使用S3File连接器连接MinIO
  S3File {
    bucket = "s3a://qq-test"
    path = "/user_json/3ebe910703c14983bb44.json"
    access_key = ""
    secret_key = "x"
    fs.s3a.endpoint = "http://10.0.10.103:9000"
    charset = "UTF-8"
    fs.s3a.aws.credentials.provider = "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider"
    file_format_type = "json"
    
    plugin_output = "raw_data" 
    # 定义复杂的嵌套 Schema
    # SeaTunnel 会根据这个 Schema 解析 JSON 内容
    schema = {
      fields {
        profile = string
        experience = string
        achievements = string
        relationships = string
        mediaAndNews = string
        uniq_id = string
      }
    }
  }
}

transform {
Sql {
    plugin_input = ["raw_data"]
    query = """
      select 
        uniq_id,
        CAST(profile AS String) as PROFILE_STR
      from raw_data
    """
    plugin_output = "stringified_data"
  }

Embedding {
    plugin_input = ["stringified_data"]
	model_provider = OPENAI
    model = bge-m3
    api_key = ''
    secret_key = ''
    api_path = "http://10.0.10.49:10003/v1/embeddings"
    vectorization_fields {
      profile_vector = PROFILE_STR
    }
    plugin_output = "embedded_table"
    result_table_name = "embedded_table"
  }
}
sink {
  Milvus {
    url = "http://10.0.10.49:19530"
    token = "root:Milvus" # 如果没设密码，默认通常是 root:Milvus 或为空
    batch_size = 1000
    plugin_input = "embedded_table"
    source_table_name = "embedded_table"
    database = "data_goverment_test"
    collection_name = "user_data"
    schema_save_mode = "ERROR_WHEN_SCHEMA_NOT_EXIST"
    collection_description  = {
      uniq_id = "uniq_id"
      PROFILE_STR = "profile_data" # 建议在 Milvus 中设为 VARCHAR 或 JSON 类型
      profile_vector = "profile_vector" # 必须yy是 FloatVector 类型
    } 
  }
}
```

### Running Command

```shell
seatunnel --config xxx.conf -m local
```

### Error Exception

```log
2026-05-11 18:03:56,962 INFO  [.s.p.d.AbstractPluginDiscovery] [main] - find connector jar and dependency for PluginIdentifier{engineType='seatunnel', pluginType='sink', pluginName='Milvus'}: [file:/E:/software/apache-seatunnel-2.3.13-bin.tar/apache-seatunnel-2.3.13-bin/apache-seatunnel-2.3.13/connectors/connector-milvus-2.3.13.jar]
2026-05-11 18:03:56,970 DEBUG [o.a.s.a.t.f.FactoryUtil       ] [main] - java.lang.NullPointerException: Cannot invoke "org.apache.seatunnel.api.table.factory.TableSinkFactoryContext.getOptions()" because "context" is null
        at org.apache.seatunnel.connectors.seatunnel.milvus.sink.MilvusSinkFactory.createSink(MilvusSinkFactory.java:56)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.lambda$createAndPrepareSink$1(FactoryUtil.java:213)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.isFallback(FactoryUtil.java:458)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSink(FactoryUtil.java:209)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.createSinkAction(MultipleTableJobConfigParser.java:690)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSink(MultipleTableJobConfigParser.java:613)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:260)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:122)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:190)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:204)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:40)

2026-05-11 18:03:56,977 INFO  [o.a.s.a.t.f.FactoryUtil       ] [main] - Create sink 'Milvus' with upstream input catalog-table[database: null, schema: null, table: raw_data]
2026-05-11 18:03:56,979 INFO  [p.MultipleTableJobConfigParser] [main] - Unsupported multi table sink api, rollback to sink template

...


2026-05-11 18:03:57,516 DEBUG [g.n.s.i.g.n.NettyClientHandler] [grpc-nio-worker-ELG-1-4] - [id: 0xe12e4b58, L:/172.20.3.24:51994 - R:/10.0.10.49:19530] INBOUND DATA: streamId=3 padding=0 endStream=false length=113 bytes=000000006c0a00125e0a0776322e342e3131121c547565205365702031302030303a31333a33352055544320323032341a07633136363235332220676f207665...
2026-05-11 18:03:57,516 DEBUG [g.n.s.i.g.n.NettyClientHandler] [grpc-nio-worker-ELG-1-4] - [id: 0xe12e4b58, L:/172.20.3.24:51994 - R:/10.0.10.49:19530] OUTBOUND PING: ack=false bytes=1234
2026-05-11 18:03:57,518 DEBUG [g.n.s.i.g.n.NettyClientHandler] [grpc-nio-worker-ELG-1-4] - [id: 0xe12e4b58, L:/172.20.3.24:51994 - R:/10.0.10.49:19530] INBOUND HEADERS: streamId=3 headers=GrpcHttp2ResponseHeaders[grpc-status: 0, grpc-message: ] padding=0 endStream=true
2026-05-11 18:03:57,525 DEBUG [g.n.s.i.g.n.NettyClientHandler] [grpc-nio-worker-ELG-1-4] - [id: 0xe12e4b58, L:/172.20.3.24:51994 - R:/10.0.10.49:19530] INBOUND PING: ack=true bytes=1234
2026-05-11 18:03:57,526 INFO  [a.s.a.s.SaveModeExecuteWrapper] [seatunnel-coordinator-service-2] - Executing save mode for table: data_goverment_test.raw_data, with SchemaSaveMode: ERROR_WHEN_SCHEMA_NOT_EXIST, DataSaveMode: APPEND_DATA using Catalog: S3File


...



2026-05-11 18:03:57,544 ERROR [o.a.s.e.s.CoordinatorService  ] [seatunnel-coordinator-service-2] - [localhost]:5801 [seatunnel-405222] [5.1] submit job 1106159750325731329 error org.apache.seatunnel.common.exception.SeaTunnelRuntimeException: ErrorCode:[API-09], ErrorDescription:[Handle save mode failed]
        at org.apache.seatunnel.engine.server.master.JobMaster.handleSaveMode(JobMaster.java:576)
        at org.apache.seatunnel.engine.server.master.JobMaster.lambda$init$1(JobMaster.java:267)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at org.apache.seatunnel.engine.server.master.JobMaster.init(JobMaster.java:261)
        at org.apache.seatunnel.engine.server.CoordinatorService.lambda$submitJob$6(CoordinatorService.java:663)
        at org.apache.seatunnel.api.tracing.MDCRunnable.run(MDCRunnable.java:43)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.apache.seatunnel.common.exception.SeaTunnelRuntimeException: ErrorCode:[API-11], ErrorDescription:[The sink table not exist] - The sink table not exist
        at org.apache.seatunnel.api.sink.DefaultSaveModeHandler.errorWhenSchemaNotExist(DefaultSaveModeHandler.java:145)
        at org.apache.seatunnel.api.sink.DefaultSaveModeHandler.handleSchemaSaveMode(DefaultSaveModeHandler.java:91)
        at org.apache.seatunnel.api.sink.SaveModeHandler.handleSaveMode(SaveModeHandler.java:42)
        at org.apache.seatunnel.api.sink.SaveModeExecuteWrapper.execute(SaveModeExecuteWrapper.java:36)
        at org.apache.seatunnel.engine.server.master.JobMaster.handleSaveMode(JobMaster.java:571)
        ... 20 more
```

### Zeta or Flink or Spark Version

zeta

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/docs/en/connectors/sink/Milvus.md
+++ b/docs/en/connectors/sink/Milvus.md
@@ -44,6 +44,7 @@ This Milvus sink connector write data to Milvus or Zilliz Cloud, it has the foll
 | url                    | String              | Yes      | -                            | The URL to connect to Milvus or Zilliz Cloud.                                                                                                       |
 | token                  | String              | Yes      | -                            | User:password                                                                                                                                       |
 | database               | String              | No       | -                            | Write data to which database, default is source database.                                                                                           |
+| collection             | String              | No       | -                            | Write data to which collection, default is source table name. The deprecated `collection_name` key is accepted as an alias.                         |
 | schema_save_mode       | enum                | No       | CREATE_SCHEMA_WHEN_NOT_EXIST | Auto create table when table not exist.                                                                                                             |
 | enable_auto_id         | boolean             | No       | false                        | Primary key column enable autoId.                                                                                                                   |
 | enable_upsert          | boolean             | No       | false                        | Upsert data not insert.                                                                                                                             |
@@ -62,6 +63,7 @@ sink {
   Milvus {
     url = "http://127.0.0.1:19530"
     token = "username:password"
+    collection = "user_vectors"
     batch_size = 1000
   }
 }

--- a/docs/zh/connectors/sink/Milvus.md
+++ b/docs/zh/connectors/sink/Milvus.md
@@ -44,6 +44,7 @@ Milvus sink连接器将数据写入Milvus或Zilliz Cloud，它具有以下功能
 | url                    | String              | 是    | -                            | 连接到Milvus或Zilliz Cloud的URL。                                         |
 | token                  | String              | 是    | -                            | 用户：密码                                                               |
 | database               | String              | 否    | -                            | 将数据写入哪个数据库，默认为源数据库。                                                 |
+| collection             | String              | 否    | -                            | 将数据写入哪个集合，默认为源表名。兼容旧配置键 `collection_name`。                              |
 | schema_save_mode       | enum                | 否    | CREATE_SCHEMA_WHEN_NOT_EXIST | 当表不存在时自动创建表。                                                        |
 | enable_auto_id         | boolean             | 否    | false                        | 主键列启用autoId。                                                        |
 | enable_upsert          | boolean             | 否    | false                        | 是否启用upsert。                                                         |
@@ -62,6 +63,7 @@ sink {
   Milvus {
     url = "http://127.0.0.1:19530"
     token = "username:password"
+    collection = "user_vectors"
     batch_size = 1000
   }
 }

--- a/seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/config/MilvusBaseOptions.java
+++ b/seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/config/MilvusBaseOptions.java
@@ -34,6 +34,7 @@ public abstract class MilvusBaseOptions {
             Options.key("collection")
                     .stringType()
                     .noDefaultValue()
+                    .withFallbackKeys("collection_name")
                     .withDescription("Milvus collection");
 
     public static final Option<String> TOKEN =

--- a/seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/MilvusSinkFactory.java
+++ b/seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/MilvusSinkFactory.java
@@ -44,15 +44,26 @@ public class MilvusSinkFactory implements TableSinkFactory {
         return OptionRule.builder()
                 .required(MilvusSinkOptions.URL, MilvusSinkOptions.TOKEN)
                 .optional(
+                        MilvusSinkOptions.DATABASE,
+                        MilvusSinkOptions.COLLECTION,
+                        MilvusSinkOptions.COLLECTION_DESCRIPTION,
+                        MilvusSinkOptions.PARTITION_KEY,
                         MilvusSinkOptions.ENABLE_UPSERT,
                         MilvusSinkOptions.ENABLE_DYNAMIC_FIELD,
                         MilvusSinkOptions.ENABLE_AUTO_ID,
+                        MilvusSinkOptions.BATCH_SIZE,
+                        MilvusSinkOptions.RATE_LIMIT,
+                        MilvusSinkOptions.LOAD_COLLECTION,
+                        MilvusSinkOptions.CREATE_INDEX,
                         MilvusSinkOptions.SCHEMA_SAVE_MODE,
                         MilvusSinkOptions.DATA_SAVE_MODE)
                 .build();
     }
 
     public TableSink createSink(TableSinkFactoryContext context) {
+        if (context == null) {
+            return () -> new MilvusSink(null, null);
+        }
         ReadonlyConfig config = context.getOptions();
         CatalogTable catalogTable = renameCatalogTable(config, context.getCatalogTable());
         return () -> new MilvusSink(config, catalogTable);

--- a/seatunnel-connectors-v2/connector-milvus/src/test/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/MilvusSinkFactoryTest.java
+++ b/seatunnel-connectors-v2/connector-milvus/src/test/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/MilvusSinkFactoryTest.java
@@ -1,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.milvus.sink;
+
+import org.apache.seatunnel.api.configuration.Option;
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
+import org.apache.seatunnel.api.configuration.util.OptionRule;
+import org.apache.seatunnel.api.sink.SeaTunnelSink;
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.Column;
+import org.apache.seatunnel.api.table.catalog.PhysicalColumn;
+import org.apache.seatunnel.api.table.catalog.TableIdentifier;
+import org.apache.seatunnel.api.table.catalog.TableSchema;
+import org.apache.seatunnel.api.table.connector.TableSink;
+import org.apache.seatunnel.api.table.factory.TableSinkFactoryContext;
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.connectors.seatunnel.milvus.config.MilvusSinkOptions;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MilvusSinkFactoryTest {
+
+    private final MilvusSinkFactory factory = new MilvusSinkFactory();
+
+    @Test
+    void testCollectionNameAliasUsesTargetCollection() {
+        SeaTunnelSink<?, ?, ?, ?> sink =
+                createSeaTunnelSink(
+                        config(
+                                "collection_name",
+                                "user_data",
+                                MilvusSinkOptions.DATABASE.key(),
+                                "data_goverment_test"));
+
+        CatalogTable writeCatalogTable = writeCatalogTable(sink);
+        assertEquals("data_goverment_test", writeCatalogTable.getTablePath().getDatabaseName());
+        assertEquals("user_data", writeCatalogTable.getTablePath().getTableName());
+    }
+
+    @Test
+    void testCollectionUsesTargetCollection() {
+        SeaTunnelSink<?, ?, ?, ?> sink =
+                createSeaTunnelSink(config(MilvusSinkOptions.COLLECTION.key(), "user_data"));
+
+        CatalogTable writeCatalogTable = writeCatalogTable(sink);
+        assertEquals("source_database", writeCatalogTable.getTablePath().getDatabaseName());
+        assertEquals("user_data", writeCatalogTable.getTablePath().getTableName());
+    }
+
+    @Test
+    void testCollectionTakesPrecedenceOverCollectionNameAlias() {
+        SeaTunnelSink<?, ?, ?, ?> sink =
+                createSeaTunnelSink(
+                        config(
+                                MilvusSinkOptions.COLLECTION.key(),
+                                "canonical_collection",
+                                "collection_name",
+                                "alias_collection"));
+
+        assertEquals("canonical_collection", writeCatalogTable(sink).getTablePath().getTableName());
+    }
+
+    @Test
+    void testCreateSinkAllowsNullContextForFallbackProbe() {
+        TableSink<?, ?, ?, ?> tableSink = assertDoesNotThrow(() -> factory.createSink(null));
+        assertNotNull(tableSink);
+    }
+
+    @Test
+    void testOptionRuleIncludesSinkOptions() {
+        List<Option<?>> declaredOptions = declaredOptions(factory.optionRule());
+
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.DATABASE));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.COLLECTION));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.BATCH_SIZE));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.PARTITION_KEY));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.COLLECTION_DESCRIPTION));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.LOAD_COLLECTION));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.CREATE_INDEX));
+        assertTrue(declaredOptions.contains(MilvusSinkOptions.RATE_LIMIT));
+    }
+
+    private SeaTunnelSink<?, ?, ?, ?> createSeaTunnelSink(Map<String, Object> options) {
+        TableSinkFactoryContext context =
+                new TableSinkFactoryContext(
+                        catalogTable(),
+                        ReadonlyConfig.fromMap(options),
+                        Thread.currentThread().getContextClassLoader());
+        return factory.createSink(context).createSink();
+    }
+
+    private CatalogTable writeCatalogTable(SeaTunnelSink<?, ?, ?, ?> sink) {
+        return sink.getWriteCatalogTable()
+                .orElseThrow(() -> new AssertionError("Expected sink write catalog table"));
+    }
+
+    private Map<String, Object> config(Object... keysAndValues) {
+        Map<String, Object> options = new HashMap<>();
+        options.put(MilvusSinkOptions.URL.key(), "http://localhost:19530");
+        options.put(MilvusSinkOptions.TOKEN.key(), "root:Milvus");
+        for (int i = 0; i < keysAndValues.length; i += 2) {
+            options.put((String) keysAndValues[i], keysAndValues[i + 1]);
+        }
+        return options;
+    }
+
+    private CatalogTable catalogTable() {
+        List<Column> columns =
+                Arrays.asList(
+                        PhysicalColumn.of(
+                                "uniq_id", BasicType.STRING_TYPE, 0L, false, null, "Unique id"),
+                        PhysicalColumn.of(
+                                "profile", BasicType.STRING_TYPE, 0L, true, null, "Profile"));
+
+        return CatalogTable.of(
+                TableIdentifier.of("S3File", "source_database", "raw_data"),
+                TableSchema.builder().columns(columns).build(),
+                new HashMap<>(),
+                Arrays.asList(),
+                "Source table");
+    }
+
+    private List<Option<?>> declaredOptions(OptionRule rule) {
+        return Stream.concat(
+                        rule.getRequiredOptions().stream()
+                                .flatMap(option -> option.getOptions().stream()),
+                        rule.getOptionalOptions().stream())
+                .collect(Collectors.toList());
+    }
+}

